inc/classes/subscriber/Media/class-webp-subscriber.php

```diff
@@ -163,14 +163,14 @@ public function convert_to_webp( $html ) {
 		$attribute_names = $this->get_attribute_names();
 
 		if ( ! $extensions || ! $attribute_names ) {
-			return $html;
+			return $html . '<!-- Rocket no webp -->';
 		}
 
 		$extensions      = implode( '|', $extensions );
 		$attribute_names = implode( '|', $attribute_names );
 
 		if ( ! preg_match_all( '@["\'\s](?<name>(?:data-(?:[a-z0-9_-]+-)?)?(?:' . $attribute_names . '))\s*=\s*["\']\s*(?<value>(?:https?:/)?/[^"\']+\.(?:' . $extensions . ')[^"\']*?)\s*["\']@is', $html, $attributes, PREG_SET_ORDER ) ) {
-			return $html;
+			return $html . '<!-- Rocket no webp -->';
 		}
```

Comment:
Is there a risk of it being processed more than once? If yes, could an edge case be that the HTML already has one of these comments appended to it, i.e. `<!-- Rocket no webp -->` or `<!-- Rocket has webp -->`?

Reply:
And on lines 134 and 139, do we need to do anything there? In those 2 cases, we are bailing out. Do we need to add the `<!-- Rocket no webp -->` to those too?

Reply:
Lines 134 and 139 are fine, because we didn't get into the browser compatibility check yet in this, and we have the necessary checks on the cache side to not have any issue.

As far as I know it doesn't happen that the methods hooked on `rocket_buffer` are executed more than once per page.

Reply:
>Lines 134 and 139 are fine, because we didn't get into the browser compatibility check yet in this, and we have the necessary checks on the cache side to not have any issue.

What happens to the cache when the page contains no images + using a WebP compatible browser +:
- `cache_webp` is off?
- or some callback is hooked into `rocket_disable_webp_cache` short-circuits the method? 

is the cache recreated for that web page?

Reply:
No in those cases, all browsers will be looking for the cache file without the `webp` suffix, so it's all good.